multipatch_analysis/matrix_analyzer/matrix_display.py

```diff
@@ -216,7 +219,10 @@ def update_matrix_display(self, results, group_results, cell_classes, cell_group
         # attempt to fix so it doesn't break in mp_a\ui\graphics.py
         # at line 90. 
         rows = []
-        for i,cell_class in enumerate(cell_classes):
+        cols = []
+        pre_names = [pre.name for pre in pre_cell_classes] ## equality comparison happens on names, so need to convert to list of names here to ask if cell is in the list later
+        post_names = [post.name for post in post_cell_classes]
+        for i,cell_class in enumerate(set(pre_cell_classes+post_cell_classes)):
```

Comment:
according to Luke this randomizes the pre/post cell_classes and may be the cause of the unorganized matrix labels